raster/src/main/scala/geotrellis/raster/VectorToRaster.scala

```diff
@@ -129,13 +135,16 @@ object VectorToRaster {
                 var ws = 0.0
                 val length = pts.size
 
-                cfor(0)(_ < length, _ + 1) { i =>
+                var i = 0
+                while (i < length) {
                   val point = pts(i)
                   val dX = (destX - point.geom.x)
                   val dY = (destY - point.geom.y)
                   val d = dX * dX + dY * dY
+                  i += 1
+
                   if (d < rr) {
-                    val w = 1 / d
+                    val w = if (d == 0) w0 else 1 / d
```

Comment:
This fixes the NoData problem for when we have a point exactly for the co-ordinates we're trying to interpolate but not sure this is totally correct. Probably we should short-circuit the interpolation and return the actual value of the point. As d tends to 0 the weight tends to infinity, so it should dominate any other values. Also currently I think its possible for a point that is slightly off the focus co-ordinates to have more weight than the actual point.